src/OpenRiaServices.Tools/Framework/CreateOpenRiaClientFilesTask.cs

```diff
@@ -788,6 +867,72 @@ internal void GenerateClientProxies()
+            {
+                var project = projectCollection.LoadProject(projectPath, projectCollection.DefaultToolsVersion);
+
+                var targetFramework = project.GetProperty("TargetFramework")?.EvaluatedValue;
+                if (targetFramework != null)
+                    return IsNetFrameworkTargetFramework(targetFramework);
+
+                var targetFrameworks = project.GetProperty("TargetFrameworks")?.EvaluatedValue;
+                if (targetFrameworks != null)
+                {
+                    // TODO: Consider choosing first framework ?
+                    if (!targetFrameworks.Contains(';'))
+                    {
+                        return IsNetFrameworkTargetFramework(targetFrameworks);
+                    }
+                    else
+                    {
+                        if (ServerAssemblies.FirstOrDefault() is ITaskItem serverAssembly)
+                        {
+                            var targetIdentifier = serverAssembly.GetMetadata("TargetFrameworkIdentifier");
+                            if (!string.IsNullOrEmpty(targetIdentifier))
+                            {
+                                bool isFramework = targetIdentifier == ".NETFramework";
+                                Log.LogMessage("Is server project .NETFramework based on TargetFrameworkIdentifier: {0}", isFramework.ToString());
+
+                                return isFramework;
+                            }
+
+                            // TODO: Look at output assembly instead of what framework the code generation uses
+                            // If ServerProject this
+                            // An other solution would also be to look at the assemblies referenced
+                            // and se if there are any paths which contains '.NETFramework' or '\net4*\'
+                            using var server = Mono.Cecil.AssemblyDefinition.ReadAssembly(serverAssembly.ItemSpec);
+
+                            var name = typeof(TargetFrameworkAttribute).FullName;
+                            var targetFrameworkAttribute =
+                                server.CustomAttributes.FirstOrDefault(a => a.AttributeType.FullName == name);
+
+                            if (targetFrameworkAttribute != null
+                                && targetFrameworkAttribute.HasConstructorArguments)
+                            {
+                                bool isFramework = targetFrameworkAttribute.ConstructorArguments[0].Value.ToString().StartsWith(".NETFramework");
+                                Log.LogMessage("Is server project .NETFramework based on TargetFrameworkAttribute: {0}", isFramework.ToString());
+                                return isFramework;
+                            }
+                        }
+                    }
+                }
+
+                return true;
+            }
+        }
+
+
+        /// <summary>
+        /// If it is a Netframework, it will not contain a dot
```

Comment:
This does not feel like a robust solution but might be the best, should look at it again